test/e2e/serial/config/config.go

```diff
@@ -67,3 +76,62 @@ func GetRteCiImage() string {
 	}
 	return nropTestCIImage
 }
+
+func CheckNodesTopology() error {
+	kconfigs, err := validator.GetKubeletConfigurationsFromTASEnabledNodes(context.TODO(), e2eclient.Client)
+	if err != nil {
+		return fmt.Errorf("error while trying to get KubeletConfigurations. error: %w", err)
+	}
+
+	var nrtList nrtv1alpha2.NodeResourceTopologyList
+
+	err = e2eclient.Client.List(context.TODO(), &nrtList)
+	if err != nil {
+		return fmt.Errorf("error while trying to get NodeResourceTopology objects. error: %w", err)
+	}
+
+	policies := []nrtv1alpha2.TopologyManagerPolicy{
+		nrtv1alpha2.SingleNUMANodeContainerLevel,
+		nrtv1alpha2.SingleNUMANodePodLevel,
+	}
+	nrts := e2enrt.FilterByPolicies(nrtList.Items, policies)
+	singleNumaNodeNumber := len(nrts)
+	if singleNumaNodeNumber < minNumberOfNodesWithRightTopology {
+		return fmt.Errorf("not enough nodes with valid policy - found %d, need at least %d", len(nrts), minNumberOfNodesWithRightTopology)
+	}
+
+	nrtsSingleNumaNodePodLevel := e2enrt.FilterTopologyManagerPolicy(nrts, nrtv1alpha2.SingleNUMANodePodLevel)
+	podLevelNumber := len(nrtsSingleNumaNodePodLevel)
+	containerLevelNumber := singleNumaNodeNumber - podLevelNumber
+
+	if podLevelNumber < minNumberOfNodesWithRightTopology && containerLevelNumber < minNumberOfNodesWithRightTopology {
+		return fmt.Errorf("not enough nodes with valid policy - found SingleNumaNodePodLevel=%d, SingleNumaNodeContainerLevel=%d, need at least %d", podLevelNumber, containerLevelNumber, minNumberOfNodesWithRightTopology)
+	}
+
+	if podLevelNumber >= minNumberOfNodesWithRightTopology {
+		for _, nrt := range nrtsSingleNumaNodePodLevel {
+			kconfig, ok := kconfigs[nrt.Name]
+			if !ok {
+				return fmt.Errorf("Unable to find KubeletConfig for node %q", nrt.Name)
+			}
+			if kconfig.TopologyManagerPolicy != "single-numa-node" || kconfig.TopologyManagerScope != "pod" {
+				return fmt.Errorf("Uncoherent KubeletConfig/NRT for node %q", nrt.Name)
```

Comment:
typo: Incoherent

other than that this check is probably better moved in the RTE testsuite, or in an explicit e2e test in general. We can and should assume _here_ we can trust the NRT data, and we can trust it correctly represents the kubelet config

Reply:
Ok, lets sync here. 
I added all the kubeletconfig stuff because I understood from [this comment](https://github.com/openshift-kni/numaresources-operator/pull/606#discussion_r1236755683) of @shajmakh that we should check both KubeletConfiguration and NRTs where coherent in this fields.
Did I understand correctly? 
Is there a possible case where the KubeletConfiguration and the NRTs had different values in these fields?

Reply:
For the sake of completeness let me summarize the agreement we have reached here
 1. keep the existing code, but fail (not skip) if the misalignment is detected
 2. file a bug/issue/jira card to track refactoring/enhancements in this area in the validation code 
 3. file a bug/issue/jira card to track adding a NROP/RTE e2e test to check this